flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecCalc.java

```diff
@@ -53,4 +55,11 @@ public BatchExecCalc(
                 outputType,
                 description);
     }
+
+    @Override
+    protected TransformationMetadata createTransformationMeta(TableConfig tableConfig) {
+        return new TransformationMetadata(
```

Comment:
instead of adding this method to many batch node. perform an instanceof check and simlar to `org.apache.flink.table.planner.plan.nodes.exec.ExecNodeContext#newContext` you call the corresponding constructor